File: examples/nodl_tutorials/navigate_to_pose_nodl/src/navigate_to_pose_client.cpp
// SPDX-FileCopyrightText: 2026 Open Source Robotics Foundation, Inc.
// SPDX-License-Identifier: Apache-2.0

#include <memory>

#include "nav2_msgs/action/navigate_to_pose.hpp"
#include "navigation.hpp"  // NOLINT(build/include_subdir)
#include "rclcpp/rclcpp.hpp"
#include "rclcpp_action/rclcpp_action.hpp"

class NavigateToPoseClient : public rclcpp::Node
{
public:
  NavigateToPoseClient()
  : Node("navigate_to_pose_client")
  , client_(rclcpp_action::create_client<nav2_msgs::action::NavigateToPose>(this, "navigate_to_pose"))
  {}

  NavigateToPoseActionClient::SharedPtr client() const
  {
    return client_;
  }

private:
  NavigateToPoseActionClient::SharedPtr client_;
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<NavigateToPoseClient>();
  const int result = run_navigation(argc, argv, node, node->client());
  node.reset();
  if (rclcpp::ok()) {
    rclcpp::shutdown();
  }
  return result;
}

File: examples/nodl_tutorials/navigate_to_pose_nodl/navigate_to_pose_nodl/navigate_to_pose_client.py
#!/usr/bin/env python3
# SPDX-FileCopyrightText: 2026 Open Source Robotics Foundation, Inc.
# SPDX-License-Identifier: Apache-2.0
"""Original action-client implementation used as the migration baseline."""

from nav2_msgs.action import NavigateToPose
from rclpy.action import ActionClient
from rclpy.node import Node

from navigate_to_pose_nodl.navigation import run


class NavigateToPoseClient(Node):
    """Create the ROS action client directly in application code."""

    def __init__(self):
        super().__init__('navigate_to_pose_client_py')
        self.action_cli_navigate_to_pose = ActionClient(self, NavigateToPose, 'navigate_to_pose')


def main(args=None):
    """Send one navigation goal with the original client."""
    return run(NavigateToPoseClient, args)


if __name__ == '__main__':
    raise SystemExit(main())
